src/Uno.UWP/Globalization/Calendar.cs

```diff
@@ -351,6 +351,10 @@ public void SetToMin()
 		public void SetToMax()
 		{
 			var calendarMaxSupportedDateTime = _calendar.MaxSupportedDateTime;
+			if (calendarMaxSupportedDateTime == DateTime.MaxValue)
```

Comment:
Can we have some explanation on why this is causing the crash and why this fix has been added (comments in code would be useful). Otherwise, when someone comes to look at this again they'll have to dig through commit history to this PR and then have no idea why it was added.
Is there any issue with the max time not being the same as MaxSupportedDateTime?

Reply:
Added a comment